daemon/src/connection.rs

```diff
@@ -0,0 +1,45 @@
+use daemon::{send_to_socket, taker_cfd, wire};
+use futures::{Stream, StreamExt};
+use std::net::SocketAddr;
+use std::thread::sleep;
+use std::time::Duration;
+use tokio_util::codec::FramedRead;
+use xtra::prelude::MessageChannel;
+use xtra::spawn::TokioGlobalSpawnExt;
+use xtra::Actor as _;
+
+const CONNECTION_RETRY_INTERVAL: Duration = Duration::from_secs(5);
+
+pub struct Actor {
```

Comment:
Any chance we can have this split out in its own commit easily?

If it is too hard, don't bother.

Reply:
not too easily, I'll keep it mind to be more atomic with my commits! (i